tests/test_cases/issue_1031/issue_1031.py

```diff
@@ -0,0 +1,24 @@
+from cocotb import test, fork
+from cocotb.triggers import Lock, Timer
+
+@test()
+async def test_trigger_lock(dut):
+
+    resource = [0]
+    lock = Lock()
+
+    fork(co(resource, lock))
+    async with lock:
+        for i in range(4):
+            resource[0] += 1
+            await Timer(10, "ns")
+    assert resource[0]==4
+    async with lock:
```

Comment:
I am not super familiar with the scheduler, but I had assumed co-routines that are blocking on an acquired lock would have priority over co-routines that attempt to acquire the same lock in the future. Is this not case?

Since I am trying to test the async context management stuff, I added an `await Timer(10, "ns")` instead to guarantee the forked co-routine acquires its lock prior to the test in this instance.

Reply:
Yeah, I agree with @andrewandrepowell here, the thing being tested for here is supposed to be guaranteed.

Reply:
@eric-wieser It is pretty atypical of a lock to make an guarantees about how it is scheduled. That is not documented. With line 24 now added however, it is valid.